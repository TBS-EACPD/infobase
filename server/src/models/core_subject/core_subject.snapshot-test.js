const crso_query = `
query CRSOTestQuery ($lang: String!){
  root(lang: $lang){
    org(org_id:"133"){
      acronym
      crsos {
        id
      }
    }
    crso(id:"ND-BUN00"){
      name
      description
      programs {
        id
      }
      org {
        id
      }
    }
    program(id:"ND-BUN05"){
      crso {
        id
      }
    }
  }
}
`;

const { execQuery } = global;

describe("subject linkage", function () {
  it("crso root, up and down linkage", async () => {
    const data = await execQuery(crso_query, {});
    return expect(data).toMatchSnapshot();
  });
});

const org_fields_query = `
query IgocTestQuery ($lang: String!){
  root(lang: $lang){
    
    org(org_id:"133"){
      id
      org_id
      dept_code
      level
      name
      legal_title
      applied_title
      old_applied_title
      acronym
      mandate
      description
    
      dp_url
      qfr_url
      eval_url
      website_url
  
      enabling_instrument
      pas_code
      faa_schedule_institutional
      faa_schedule_hr_status
      auditor
      incorp_yr
      federal_ownership
      end_yr
      notes
      dp_status
      article1_fr
      article2_fr

      inst_form {
        id
      }
      ministry {
        name
        id
        orgs {
          id
        }
      }
      ministers {
        id
        name
      }

      programs {
        id
      }
      crsos {
        id
      }
    }
    org_with_notes: org(org_id: "326"){
      notes
    }
    org_with_auditor: org(org_id: "15"){
      auditor
    }
    org_with_end_date: org(org_id:"326"){
      end_yr
    }
    org_with_faa_hr: org(org_id:"552"){
      faa_schedule_hr_status
    }
    org_with_fed_ownership: org(org_id: "321"){
      federal_ownership
    }
  }
}
`;

describe("igoc linkage and fields", function () {
  it("igoc linkage and fields", async () => {
    const data = await execQuery(org_fields_query, {});
    return expect(data).toMatchSnapshot();
  });
});

const program_query = `
query ProgramTestQuery ($lang: String!){
  root(lang: $lang){
    program(id: "ND-ATT00"){
      id
      activity_code
      is_internal_service
      description
      
    }
    internal_service_program: program(id:"TBC-BXD01"){
      is_internal_service
    }
    inactive_program: program(id: "TBC-DAC00"){
      is_active
    }
    
  }
}
`;

describe("program basic fields", function () {
  it("program basic fields", async () => {
    const data = await execQuery(program_query, {});
    return expect(data).toMatchSnapshot();
  });
});
