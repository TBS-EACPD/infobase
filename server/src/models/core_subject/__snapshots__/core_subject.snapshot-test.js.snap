// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`igoc linkage and fields igoc linkage and fields 1`] = `
Object {
  "data": Object {
    "root": Object {
      "org": Object {
        "acronym": "DND",
        "applied_title": null,
        "article1_fr": null,
        "article2_fr": "de",
        "auditor": null,
        "crsos": Array [
          Object {
            "id": "ND-BUN00",
          },
          Object {
            "id": "ND-BUO00",
          },
          Object {
            "id": "ND-BUP00",
          },
          Object {
            "id": "ND-BUQ00",
          },
          Object {
            "id": "ND-BUR00",
          },
          Object {
            "id": "ND-BUS00",
          },
          Object {
            "id": "ND-ISS00",
          },
          Object {
            "id": "ND-SO217",
          },
          Object {
            "id": "ND-SO218",
          },
          Object {
            "id": "ND-SO219",
          },
          Object {
            "id": "ND-SO220",
          },
          Object {
            "id": "ND-SO221",
          },
          Object {
            "id": "ND-SO222",
          },
          Object {
            "id": "ND-SO223",
          },
        ],
        "dept_code": "ND",
        "description": "On behalf of the people of Canada, the Canadian Armed Forces (CAF) and the Department of National Defence (DND) stand ready to perform three key roles:Protecting Canada and defending our sovereignty;Defending North America in cooperation with the United States, Canada's closest ally; andContributing to international peace and security through operations around the world, most often in partnership with allies from other countries.",
        "dp_status": "0",
        "enabling_instrument": "National Defence Act, R.S.C. 1985, c. N-5",
        "end_yr": null,
        "eval_url": null,
        "faa_schedule_hr_status": null,
        "faa_schedule_institutional": "I",
        "federal_ownership": null,
        "id": "133",
        "incorp_yr": "1923",
        "inst_form": Object {
          "id": null,
        },
        "legal_title": null,
        "mandate": "On behalf of the people of Canada, the Canadian Armed Forces (CAF) and the Department of National Defence (DND) stand ready to perform three key roles:Protecting Canada and defending our sovereignty;Defending North America in cooperation with the United States, Canada's closest ally; andContributing to international peace and security through operations around the world, most often in partnership with allies from other countries.",
        "ministers": Array [
          Object {
            "id": null,
            "name": null,
          },
        ],
        "ministry": Object {
          "id": null,
          "name": null,
          "orgs": null,
        },
        "name": "National Defence",
        "notes": null,
        "old_applied_title": null,
        "org_id": "133",
        "pas_code": "DND",
        "programs": Array [
          Object {
            "id": "ND-BUN01",
          },
          Object {
            "id": "ND-BUN02",
          },
          Object {
            "id": "ND-BUN03",
          },
          Object {
            "id": "ND-BUN04",
          },
          Object {
            "id": "ND-BUN05",
          },
          Object {
            "id": "ND-BUN06",
          },
          Object {
            "id": "ND-BUN07",
          },
          Object {
            "id": "ND-BUO01",
          },
          Object {
            "id": "ND-BUO02",
          },
          Object {
            "id": "ND-BUO03",
          },
          Object {
            "id": "ND-BUO04",
          },
          Object {
            "id": "ND-BUO05",
          },
          Object {
            "id": "ND-BUO06",
          },
          Object {
            "id": "ND-BUO07",
          },
          Object {
            "id": "ND-BUO08",
          },
          Object {
            "id": "ND-BUO09",
          },
          Object {
            "id": "ND-BUO10",
          },
          Object {
            "id": "ND-BUO11",
          },
          Object {
            "id": "ND-BUP01",
          },
          Object {
            "id": "ND-BUP02",
          },
          Object {
            "id": "ND-BUP03",
          },
          Object {
            "id": "ND-BUP04",
          },
          Object {
            "id": "ND-BUP05",
          },
          Object {
            "id": "ND-BUP06",
          },
          Object {
            "id": "ND-BUP07",
          },
          Object {
            "id": "ND-BUP08",
          },
          Object {
            "id": "ND-BUP09",
          },
          Object {
            "id": "ND-BUP10",
          },
          Object {
            "id": "ND-BUQ01",
          },
          Object {
            "id": "ND-BUQ02",
          },
          Object {
            "id": "ND-BUQ03",
          },
          Object {
            "id": "ND-BUQ04",
          },
          Object {
            "id": "ND-BUQ05",
          },
          Object {
            "id": "ND-BUQ06",
          },
          Object {
            "id": "ND-BUQ07",
          },
          Object {
            "id": "ND-BUQ08",
          },
          Object {
            "id": "ND-BUR01",
          },
          Object {
            "id": "ND-BUR02",
          },
          Object {
            "id": "ND-BUR03",
          },
          Object {
            "id": "ND-BUR04",
          },
          Object {
            "id": "ND-BUR05",
          },
          Object {
            "id": "ND-BUS01",
          },
          Object {
            "id": "ND-BUS02",
          },
          Object {
            "id": "ND-BUS03",
          },
          Object {
            "id": "ND-BUS04",
          },
          Object {
            "id": "ND-BUS05",
          },
          Object {
            "id": "ND-BUS06",
          },
          Object {
            "id": "ND-BUS07",
          },
          Object {
            "id": "ND-BUS08",
          },
          Object {
            "id": "ND-BUS09",
          },
          Object {
            "id": "ND-BUS10",
          },
          Object {
            "id": "ND-BUS11",
          },
          Object {
            "id": "ND-BUS12",
          },
          Object {
            "id": "ND-BUS13",
          },
          Object {
            "id": "ND-ISS01",
          },
          Object {
            "id": "ND-ISS02",
          },
          Object {
            "id": "ND-ISS03",
          },
          Object {
            "id": "ND-ISS04",
          },
          Object {
            "id": "ND-ISS05",
          },
          Object {
            "id": "ND-ISS06",
          },
          Object {
            "id": "ND-ISS07",
          },
          Object {
            "id": "ND-ISS08",
          },
          Object {
            "id": "ND-ISS09",
          },
          Object {
            "id": "ND-ISS0Z",
          },
          Object {
            "id": "ND-AVI00",
          },
          Object {
            "id": "ND-AVJ00",
          },
          Object {
            "id": "ND-AVM00",
          },
          Object {
            "id": "ND-AVR00",
          },
          Object {
            "id": "ND-CQA00",
          },
          Object {
            "id": "ND-CQB00",
          },
          Object {
            "id": "ND-CQC00",
          },
          Object {
            "id": "ND-CQD00",
          },
          Object {
            "id": "ND-CQE00",
          },
          Object {
            "id": "ND-AUW00",
          },
          Object {
            "id": "ND-AUX00",
          },
          Object {
            "id": "ND-AUY00",
          },
          Object {
            "id": "ND-AUZ00",
          },
          Object {
            "id": "ND-ISC00",
          },
          Object {
            "id": "ND-AUN00",
          },
          Object {
            "id": "ND-AUO00",
          },
          Object {
            "id": "ND-AUR00",
          },
          Object {
            "id": "ND-AUS00",
          },
          Object {
            "id": "ND-ATT00",
          },
          Object {
            "id": "ND-ATU00",
          },
          Object {
            "id": "ND-ATW00",
          },
          Object {
            "id": "ND-AUG00",
          },
        ],
        "subject_type": "org",
        "website_url": "www.greatlakes-seaway.com",
      },
      "org_with_auditor": Object {
        "auditor": "Ernst & Young LLP and PricewaterhouseCoopers LLP",
      },
      "org_with_end_date": Object {
        "end_yr": null,
      },
      "org_with_faa_hr": Object {
        "faa_schedule_hr_status": "V",
      },
      "org_with_fed_ownership": Object {
        "federal_ownership": "7/9 board members",
      },
      "org_with_notes": Object {
        "notes": null,
      },
    },
  },
}
`;

exports[`program basic fields program basic fields 1`] = `
Object {
  "data": Object {
    "root": Object {
      "inactive_program": Object {
        "is_active": false,
      },
      "internal_service_program": Object {
        "is_internal_service": false,
      },
      "program": Object {
        "activity_code": "ATT00",
        "description": "This program provides the Government of Canada with critical scientific knowledge and innovation to address defence & security challenges and needs. The S&T Program includes direction setting, program planning, program management, and capability management, execution and assessment. The Program comprises multi-year projects with activities in research, technology development, analysis and experimentation applied to inform, enable and respond to Canada's defence and security priorities over multiple time horizons extending up to a 20-year outlook. The scientific knowledge and innovation generated from these activities informs decisions on Defence capability acquisitions, readiness preparation and the conduct of operations in response to Government priorities. Activities under this program draw on internal capability and make extensive use of partnerships with Canadian industry and academia as well as international organizations.",
        "id": "ND-ATT00",
        "is_internal_service": false,
      },
    },
  },
}
`;

exports[`subject linkage crso root, up and down linkage 1`] = `
Object {
  "data": Object {
    "root": Object {
      "crso": Object {
        "description": "Detect, deter and defend against threats to or attacks on Canada. Assist civil authorities and law enforcement, including counter-terrorism, in support of national security, domestic disasters or major emergencies, and conduct search and rescue operations. Detect, deter and defend against threats to or attacks on North America in partnership with the United States, including through NORAD. Lead and/or contribute forces to NATO and coalition efforts to deter and defeat adversaries, including terrorists, to support global stability. Lead and/or contribute to international peace operations and stabilization missions with the United Nations, NATO and other multilateral partners. Engage in capacity building to support the security of other nations and their ability to contribute to security and the security of Canadians abroad. Assist civil authorities and non-governmental partners in responding to international and domestic disasters or major emergencies.",
        "name": "Operations",
        "org": Object {
          "id": "133",
        },
        "programs": Array [
          Object {
            "id": "ND-BUN01",
          },
          Object {
            "id": "ND-BUN02",
          },
          Object {
            "id": "ND-BUN03",
          },
          Object {
            "id": "ND-BUN04",
          },
          Object {
            "id": "ND-BUN05",
          },
          Object {
            "id": "ND-BUN06",
          },
          Object {
            "id": "ND-BUN07",
          },
        ],
      },
      "org": Object {
        "acronym": "DND",
        "crsos": Array [
          Object {
            "id": "ND-BUN00",
          },
          Object {
            "id": "ND-BUO00",
          },
          Object {
            "id": "ND-BUP00",
          },
          Object {
            "id": "ND-BUQ00",
          },
          Object {
            "id": "ND-BUR00",
          },
          Object {
            "id": "ND-BUS00",
          },
          Object {
            "id": "ND-ISS00",
          },
          Object {
            "id": "ND-SO217",
          },
          Object {
            "id": "ND-SO218",
          },
          Object {
            "id": "ND-SO219",
          },
          Object {
            "id": "ND-SO220",
          },
          Object {
            "id": "ND-SO221",
          },
          Object {
            "id": "ND-SO222",
          },
          Object {
            "id": "ND-SO223",
          },
        ],
      },
      "program": Object {
        "crso": Object {
          "id": "ND-BUN00",
        },
      },
    },
  },
}
`;
